### PR TITLE
Feat/30/add task page

### PR DESCRIPTION
**Pull request by OdanetsLiliya**

### CODE REVIEW

File: src/app/layout-navigation/layout-navigation.component.html [hunk 1]
```diff
@@ -6,5 +6,5 @@
       </mat-list-item>
     </mat-nav-list>
   </mat-drawer>
-  <exc-task-sending-page></exc-task-sending-page>
+  <exc-task-creation-page class="task-creation-page"></exc-task-creation-page>
```
Comment (annakisel) [inline src/app/layout-navigation/layout-navigation.component.html]: delete it pls

File: src/app/task-creation-page/task-creation-page.component.scss [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  min-width: 150px;
+  max-width: 600px;
+  width: 100%;
+}
+
+.full-width {
+  width: 100%;
+}
+
+textarea {
```
Comment (annakisel) [inline src/app/task-creation-page/task-creation-page.component.scss]: styles should be applied to classes

File: src/app/app.module.ts [hunk 3]
```diff
@@ -46,16 +49,20 @@ import { TaskFullDescrTeacherComponent } from './task-full-descr-teacher/task-fu
...
     TaskSendingPageComponent,
     AllTasksTeacherComponent,
     TableMyTasksTeacherComponent,
     TableMyTasksTeacherComponent,
+    InputOutputAddingComponent,
     StudentTasksTableComponent,
     AssignATaskComponent,
     SelectStudentOrGroupComponent,
     TaskFullDescrTeacherComponent,
+
```
Comment (annakisel) [inline src/app/app.module.ts]: do u need new line ?)

File: src/app/file-uploading/file-uploading.component.ts [hunk 4]
```diff
@@ -1,17 +1,23 @@
-import { Component, Output, EventEmitter } from '@angular/core';
+import { Component, Output, EventEmitter, Input } from '@angular/core';
+
```
Comment (annakisel) [inline src/app/file-uploading/file-uploading.component.ts]: the same

File: src/app/file-uploading/file-uploading.component.ts [hunk 5]
```diff
@@ -1,17 +1,23 @@
...
 
 @Component({
   selector: 'exc-file-uploading',
   templateUrl: './file-uploading.component.html',
   styleUrls: ['./file-uploading.component.scss']
 })
-
 export class FileUploadingComponent {
   @Output() fileChanged = new EventEmitter<File>();
+  @Input() id: number;
```
Comment (annakisel) [inline src/app/file-uploading/file-uploading.component.ts]: Will be better Input before Output, no?

File: src/app/input-output-adding/input-output-adding.component.scss [hunk 6]
```diff
@@ -0,0 +1,9 @@
+.input-output-files-upload {
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+}
+
+.file-uploading {
```
Comment (annakisel) [inline src/app/input-output-adding/input-output-adding.component.scss]: why do u use styles to component ?

File: src/app/input-output-adding/input-output-adding.component.ts [hunk 7]
```diff
@@ -0,0 +1,55 @@
+import { Component, Output, EventEmitter, Input } from '@angular/core';
+
+@Component({
+  selector: 'exc-input-output-adding',
+  templateUrl: './input-output-adding.component.html',
+  styleUrls: ['./input-output-adding.component.scss']
+})
+export class InputOutputAddingComponent {
+  @Input() files: File[] = [];
```
Comment (annakisel) [inline src/app/input-output-adding/input-output-adding.component.ts]: at first Inputs, then Outputs, and then other variables)

File: src/app/input-output-adding/input-output-adding.component.ts [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+  validFile = false;
+  @Output() fileChanged = new EventEmitter<File>();
+  @Output() validFiles = new EventEmitter<boolean>();
+  length: number[] = [];
+  counter: number;
```
Comment (annakisel) [inline src/app/input-output-adding/input-output-adding.component.ts]: new line after it

File: src/app/input-output-adding/input-output-adding.component.ts [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+    // console.log(this.files);
+    this.length.push(++this.counter);
+    this.length.push(++this.counter);
+    this.files.length += 2;
+    this.validFile = false;
+    this.validFiles.emit(this.validFile);
+  }
+
+  onDelete(event, index) {
+    // console.log(index);
```
Comment (annakisel) [inline src/app/input-output-adding/input-output-adding.component.ts]: would be better to delete this commets

File: src/app/task-creation-page/task-creation-page.component.html [hunk 10]
```diff
@@ -0,0 +1,12 @@
+<form class="form" #form="ngForm" (ngSubmit)="onSubmit(form.value)">
+  <mat-form-field class="full-width">
+    <input matInput class="input" #name ngModel name="name" maxlength="20" placeholder="Название" required>
+    <mat-hint align="end">{{name.value.length}} / 20</mat-hint>
+  </mat-form-field>
+
+  <mat-form-field class="full-width">
+    <textarea class='description-textarea' matInput placeholder="Описание" rows="10" ngModel name='description' required></textarea>
```
Comment (annakisel) [inline src/app/task-creation-page/task-creation-page.component.html]: must be double quotes in html files

File: src/app/file-uploading/file-uploading.component.ts (hunk 5 above)
Comment (OdanetsLiliya) [reply src/app/file-uploading/file-uploading.component.ts]: why?

**Pull request merged**